Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Payment Flow/PaymentFormViewController.m

```diff
@@ -75,6 +59,31 @@ - (void)pay {
     }];
 }
 
+- (void)completePayment {
+    //FIXME: Show loading refreshing control
+    BasketTransaction *basketTx = [BasketTransaction new];
+    basketTx.basketRecipient = self.basket;
+    basketTx.madeByUser = [User currentUser];
+    basketTx.totalAmount = self.totalAmount;
+    [self.basket.allTransactions addObject:basketTx];
+    self.basket.featuredValueDict[@"numberOfDonations"] = [[NSNumber alloc] initWithLong:self.basket.allTransactions.count];
```

Comment:
> You can try to mitigate this by refreshing the basket right before the update, but won't be 100% accurate either.

Hm, so what's the best way to get that value and update it? It seems like refreshing & updating is the best thing to do here. Thanks for thinking of this.

Reply:
> Also why do you need to store the numberOfDonations here if the information is already available in basket.allTransactions.count?

I'm imagining that the logic for the featuredValue takes into account the number of transactions to that basket, but will not actually be == basket.allTransactions.count.

Reply:
I'm not talking about the featuredValue, I'm talking about `numberOfDonations` in particular, isn't that exactly the same as `basket.allTransactions.count`?

Reply:
> Hm, so what's the best way to get that value and update it?

Usually this would be done on the server, in most cases the count could just be a computed value when fetching a basket. If it needs to be stored it would be just for performance reasons, and in this case the stored count would be updated on the backend on the same transaction where the new donation is added to the basket.

Reply:
I'm not sure I'm understanding this -- right now, I only fetch the basket once when loading the initial feed though & pass that basket around everywhere. Does this mean I should re-fetch the basket when I'm liking/unliking a basket?

Reply:
Re: `numberOfDonations` vs `basket.allTransactions.count` : 
They're essentially the same, but numberOfDonations is a weight that's capped at 30 (I implemented this code in [a later PR](https://github.com/kristiehuang/Basket-Donation-Payments/pull/42/commits/2a4b15746cd1a2cf8cb486dc6c317a49d6ac34c2) in the `updateBasketFeaturedValueWeights` function.)

Reply:
> I'm not sure I'm understanding this -- right now, I only fetch the basket once when loading the initial feed though & pass that basket around everywhere.
> 
> Does this mean I should re-fetch the basket when I'm liking/unliking a basket? Parse has a `fetch` function I can use for this to "Synchronously fetch the PFObject with the current data from the server"

What you have implemented will work for the purpose of this project, as long as it's just one user using the app at a time from a single device. I just wanted to point that this wouldn't work in the real world at the scale of a big company like Facebook, where many users are using the app the same time and some users have more than one session on different devices at the same time. But I think for the purpose of this project you can leave it as is.